src/python_inspector/resolve_cli.py

```diff
@@ -369,15 +377,20 @@ def resolve(
 
 
 def get_requirements_from_direct_dependencies(
-    direct_dependencies: List[DependentPackage],
+    direct_dependencies: List[DependentPackage], environment_marker: Dict = {}
```

Comment:
You cannot use `environment_marker: Dict = {}` as this is implicitly a shared global variable